### DISCUSSION TITLE
How to duplicate ‘/Root’, except ‘/Metadata’, from PdfReader to PdfWriter?

### DISCUSSION THREAD

**Discussion by macdeport** [py-pdf/pypdf#2688] [Q&A]
`create_viewer_preferences()` exists, but how do you define these preferences?

**Comment by pubpub-zz** [comment]
https://pypdf.readthedocs.io/en/stable/user/viewer-preferences.html?highlight=create_viewer_preferences

**Reply by macdeport** [reply]
@pubpub-zz Thank you.

But in fact I have to clone `PdfReader` `viewer_preferences` in `PdfWriter`: what is the best way to do?

More generally I try to duplicate ‘/Root’, except ‘/Metadata’, from PdfReader to PdfWriter.

**Reply by pubpub-zz** [reply]
if you want just viewer_preference from reader,  You should use:
`writer.root_object[pypdf.generic.NameObject("/ViewerPreferences")] = reader.root_object["/ViewerPreferences"].clone(writer)`

**Reply by macdeport** [reply]
> if you want just viewer_preference from reader, You should use:
> `writer.root_object[pypdf.generic.NameObject("/ViewerPreferences")] = reader.root_object["/ViewerPreferences"].clone(writer)`

Works like a charm, even slightly modified:
```Python
from pypdf import generic,PdfReader, PdfWriter

reader = PdfReader(pdf_in_path); writer = PdfWriter(pdf_out_path)
writer.root_object[generic.NameObject('/ViewerPreferences')] = reader.viewer_preferences

```
@pubpub-zz  Thank you

PS: Your answer works perfectly without `.clone(writer)`
